src/tests/ftest/nvme/nvme_pool_extend.py

```diff
@@ -0,0 +1,242 @@
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import time
+import uuid
+import threading
+
+from itertools import product
+from avocado import fail_on
+from apricot import TestWithServers, skipForTicket
+from test_utils_pool import TestPool
+from ior_utils import IorCommand
+from job_manager_utils import Mpirun
+from write_host_file import write_host_file
+from command_utils import CommandFailure
+from mpio_utils import MpioUtils
+
+try:
+    # python 3.x
+    import queue as queue
+except ImportError:
+    # python 2.7
+    import Queue as queue
+
+
+class NvmePoolExtend(TestWithServers):
+    # pylint: disable=too-many-ancestors
+    """
+    Test Class Description: This test runs
+    NVME Pool Extend test cases.
+
+    :avocado: recursive
+    """
+    def setUp(self):
+        """Set up for test case."""
+        super(NvmePoolExtend, self).setUp()
+        self.dmg_command = self.get_dmg_command()
+        self.ior_w_flags = self.params.get("write_flags", '/run/ior/iorflags/*')
+        self.ior_r_flags = self.params.get("read_flags", '/run/ior/iorflags/*')
+        self.ior_apis = self.params.get("ior_api", '/run/ior/iorflags/*')
+        self.ior_test_sequence = self.params.get("ior_test_sequence",
+                                                 '/run/ior/iorflags/*')
+        self.ior_daos_oclass = self.params.get("obj_class",
+                                               '/run/ior/iorflags/*')
+        self.ior_dfs_oclass = self.params.get(
+            "obj_class", '/run/ior/iorflags/*')
+        # Start an additional server.
+        self.extra_servers = self.params.get("test_servers",
+                                             "/run/extra_servers/*")
+        # Recreate the client hostfile without slots defined
+        self.hostfile_clients = write_host_file(
+            self.hostlist_clients, self.workdir, None)
+        self.pool = None
+        self.out_queue = queue.Queue()
+        self.container_info = {}
+
+    @fail_on(CommandFailure)
+    def get_pool_version(self):
+        """Get the pool version.
+
+        Returns:
+            int: pool_version_value
+
+        """
+        data = self.dmg_command.pool_query(self.pool.uuid)
+        return int(data["version"])
+
+    def ior_thread(self, pool, oclass, api, test, flags, results):
+        """Start threads and wait until all threads are finished.
+        Args:
+            pool (object): pool handle
+            oclass (str): IOR object class
+            API (str): IOR API
+            test (list): IOR test sequence
+            flags (str): IOR flags
+            results (queue): queue for returning thread results
+        """
+        processes = self.params.get("slots", "/run/ior/clientslots/*")
+        mpio_util = MpioUtils()
+        if mpio_util.mpich_installed(self.hostlist_clients) is False:
+            self.fail("Exiting Test: Mpich not installed")
+        self.pool = pool
```

Comment:
It looks like this argument `pool` isn't necessary because you pass in self.pool to it in line 150, so it would work the same way without it.

Reply:
This is needed when handling multiple pools. You are overriding self.pool with the another pool object.

Reply:
Let me try something on my code and run a manual run and see whether I can come with a new logic to run with a single while loop for each ior data sequence.